docs/design/enclave-tls/design.md

```diff
@@ -0,0 +1,364 @@
+  - Enclave TLS client connects to the remote Enclave TLS server. Enclave TLS server listens to the TLS port.
+
+![enclave\_tls\_negotitate.png](enclave_tls_negotiate.png)
+
+3. After the Enclave-TLS trusted channel is successfully established, the client and server can directly transmit sensitive data through the Enclave TLS API `enclave_tls_transmit()` and `enclave_tls_receive()`.
+
+![enclave\_tls\_transmite.png](enclave_tls_transmite.png)
+
+4. The Enclave TLS application calls the Enclave TLS API `enclave_tls_cleanup()` to clean up the Enclave TLS operating environment. 
+As shown in the following figure: Enclave TLS will sequentially call the `clean_up()` method of Crypto Wrapper instance, Enclave Attester instance, Enclave Verifier instance, and TLS Wrapper instance to clean up the corresponding instance context (for example: close the handle, etc.), and then proceed to the core layer context The environment is cleared.
+
+![enclave\_tls\_cleanup.png](enclave_tls_cleanup.png)
+
+# Instances design
+
+According to the respective responsibilities and constraints of each instance, we have designed common metadata and APIs for each type of instance.
+
+## TLS Wrapper instance
+
+Each TLS Wrapper instance is designed with the following metadata and API to complete the functions of TLS session management and network transmission. The following is the meaning of each metadata and method.
+
+```c
+typedef struct {
+    uint8_t api_version;
+	unsigned long flags;
+	const char name[TLS_TYPE_NAME_SIZE];
+	uint8_t priority; // The higher the value, the higher the priority.
+
+	// Detect whether the current TLS library can run in the current environment
+	tls_wrapper_err_t (*pre_init)(void);
+	// Initialize the TLS library
+	tls_wrapper_err_t (*init)(tls_wrapper_ctx_t *ctx);
+	// Load the private key into the specific ssl context
+	tls_wrapper_err_t (*use_privkey)(tls_wrapper_ctx_t *ctx, void *privkey_buf,
+					 size_t privkey_len);
+	// Load the certificate buffer into a specific SSL context
+	tls_wrapper_err_t (*use_cert)(tls_wrapper_ctx_t *ctx, enclave_tls_cert_info_t *cert_info);
+	// Establish the actual TLS connection
+	tls_wrapper_err_t (*negotiate)(tls_wrapper_ctx_t *ctx, int fd);
+	// Transmiting data
+	tls_wrapper_err_t (*transmit)(tls_wrapper_ctx_t *ctx, void *buf, size_t *buf_size);
+	// Receiving data
+	tls_wrapper_err_t (*receive)(tls_wrapper_ctx_t *ctx, void *buf, size_t *buf_size);
+	// Clean up TLS library
+	tls_wrapper_err_t (*cleanup)(tls_wrapper_ctx_t *ctx);
+} tls_wrapper_opts_t;
+```
+
+`libtls_wrapper_<type>.so` will register metadata and methods to the Enclave-TLS core when it is initialized.
+
+```c
+tls_wrapper_err_t tls_wrapper_register(const tls_wrapper_opts_t *opts);
+```
+
+## Enclave Attester instance
+
+Each Attester instance has designed the following metadata and API to be responsible for collecting evidence from the local platform operating environment. The following is the meaning of each metadata and method.
```

Comment:
change Attester instance has designed the -> Attester instance is designed with ?
To align with follow "Enclave Verifier instance" description.